go-controller/pkg/ovn/ovn.go

```diff
@@ -325,6 +327,8 @@ func (oc *Controller) Run(ctx context.Context, wg *sync.WaitGroup) error {
 
 	// WatchNetworkPolicy depends on WatchPods and WatchNamespaces
 	oc.WatchNetworkPolicy()
+	// Clean up stale L4 network policies
+	oc.CleanStaleNetworkPolicy()
```

Comment:
why is this after watch network policy? Is it so that the network policy adds will be processed and update/add new acls. In other words there wont be a gap of time where network policy isn't enforced if we cleaned the stale ones first?

If this is the case, then we should probably add something to the watchResource function to take postSyncFunctions or something like that. That way we can make this a generic function that all handlers could utilize. Additionally it could have logic to retry so that your sync function can safely return an error and during a transient OVN outage we would retry again. wdyt?

@ricky-rav @flavio-fernandes

Reply:
@trozet since we want the sync function to be executed successfully before running `cleanStaleNetworkPolicy`, would it make sense to just insert it at the very end of `syncNetworkPolicies`? This way it would be called only after existing network policies are added and, in case of failure, it would be retried.  Or would you prefer to call it from `WatchResource` in `postSyncFunctions`, as you are suggesting?

Reply:
syncNetworkPolicies wont add existing network policies so I think it would need to be a postSyncFunction